test/NuGet.Core.Tests/NuGet.Commands.Test/IncludeTypeTests.cs

```diff
@@ -1759,6 +1762,106 @@ public async Task IncludeType_DependenciesWithTargetsAsync()
+                    .WithTestRestoreMetadata();
+
+                var project2Spec = PackageReferenceSpecBuilder.Create("Project2", project2Directory.FullName)
+                    .WithTargetFrameworks(new[]
+                    {
+                        new TargetFrameworkInformation
+                        {
+                            FrameworkName = NuGetFramework.Parse("net471"),
+                            Dependencies = new List<LibraryDependency>(),
+                            CentralPackageVersions = { new KeyValuePair<string, CentralPackageVersion>("PackageA", new CentralPackageVersion("PackageA", VersionRange.Parse("1.0.0"))) },
+                        }
+                    })
+                    .WithCentralPackageVersionsEnabled()
+                    .WithCentralPackageTransitivePinningEnabled()
+                    .Build()
+                    .WithTestRestoreMetadata()
+                    .WithTestProjectReference(project3Spec, privateAssets:(LibraryIncludeFlags.All & (~includeFlags)));
+
+
+                var project1Spec = PackageReferenceSpecBuilder.Create("Project1", project1Directory.FullName)
+                    .WithTargetFrameworks(new[]
+                    {
+                        new TargetFrameworkInformation
+                        {
+                            FrameworkName = NuGetFramework.Parse("net471"),
+                            Dependencies = new List<LibraryDependency>(),
+                            CentralPackageVersions = { new KeyValuePair<string, CentralPackageVersion>("PackageA", new CentralPackageVersion("PackageA", VersionRange.Parse("1.0.0"))) },
+                        }
+                    })
+                    .WithCentralPackageVersionsEnabled()
+                    .WithCentralPackageTransitivePinningEnabled()
+                    .Build()
+                    .WithTestRestoreMetadata()
+                    .WithTestProjectReference(project2Spec);
+
+                var restoreContext = new RestoreArgs()
+                {
+                    Sources = new List<string>() { packageSource.FullName },
+                    GlobalPackagesFolder = globalPackages.FullName,
+                    Log = logger,
+                    CacheContext = new TestSourceCacheContext(),
+                };
+
+                await SimpleTestPackageUtility.CreatePackagesAsync(
+                    packageSource.FullName,
+                    packageA);
+
+                var request = await ProjectTestHelpers.GetRequestAsync(restoreContext, project1Spec, project2Spec, project3Spec);
+
+                // Act
+                var command1 = new RestoreCommand(request);
+                var result1 = await command1.ExecuteAsync();
+                var lockFile1 = result1.LockFile;
+
+                // Assert
+                Assert.True(result1.Success);
+                Assert.Equal(includeFlags, lockFile1.CentralTransitiveDependencyGroups.Single().TransitiveDependencies.Single().IncludeType);
```

Comment:
This should affect the files selected as well correct?

Can we validate the targets section is correct as well?

Reply:
done 👍